modules/Database/Arangojs.js

```diff
@@ -904,6 +904,19 @@ class ArangoJS {
         });
     }
 
+    /**
+     * Returns data creator identity for dataset ID
+     * @param datasetId
+     * @returns {Promise}
+     */
+    async findIssuerIdentityForDatasetId(datasetId) {
+        const queryString = `for d in ot_datasets
```

Comment:
You could avoid iterating through the database by using `document('ot_datasets', @datasetId)`